src/main/java/frc/robot/commands/BalanceCommand.java

```diff
@@ -0,0 +1,64 @@
+package frc.robot.commands;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.constants.DriveConstants;
+import frc.robot.subsystems.Drivetrain;
+
+public class BalanceCommand extends CommandBase {
+
+    private Drivetrain m_drive;
+    private final PIDController m_pid;
+
+    private double m_currentAngle, m_output;
+
+    private boolean m_usePitch;
+
+    private int m_inverted;
+ 
+    public BalanceCommand(Drivetrain drive) {
+        m_drive = drive;
+        addRequirements(drive);
+        m_pid = new PIDController(DriveConstants.kBalanceP, DriveConstants.kBalanceI, DriveConstants.kBalanceD);
+        m_pid.setTolerance(DriveConstants.kBalanceTolerance);
+    }
+
+    @Override
+    public void initialize() {
+        m_inverted = 1;
+
+        m_pid.setSetpoint(0);
+        if(Math.abs(Math.PI/2 - m_drive.getAngleHeading() % Math.PI) > Math.PI/4)  { //Determines whether to use roll or pitch
+            m_usePitch = false;
+        }
+        else  {
+            m_usePitch = true;
+        }
+        if(m_drive.getAngleHeading() > Math.PI)
```

Comment:
The getHeaddingAngle method is modulus and will only output values from -PI to PI so there is no situation in wich a value greater than PI will be output. I think you will have to change it to value<0